core/geo-json/src/test/java/org/eclipse/sensinact/gateway/geojson/GeoJsonTest.java

```diff
@@ -161,17 +163,25 @@ void testNaN() throws Exception {
                 new TypeReference<List<Geometry>>() {
                 });
 
-        assertEquals(2, geometries.size());
+        assertEquals(1, geometries.size());
         assertEquals(Point, geometries.get(0).type);
         Point nan = (Point) geometries.get(0);
         assertTrue(Double.isNaN(nan.coordinates.latitude));
         assertTrue(Double.isNaN(nan.coordinates.longitude));
         assertTrue(Double.isNaN(nan.coordinates.elevation));
-        assertEquals(Point, geometries.get(1).type);
```

Comment:
This looks like something that should be moved into another test where we assert the behaviour on the invalid (i.e. empty) coordinates array